cdap-api/src/main/java/co/cask/cdap/internal/api/DefaultDatasetConfigurer.java

```diff
@@ -50,21 +53,43 @@
   }
 
   public void addStreams(Map<String, StreamSpecification> streams) {
+    Set<String> duplicateStreams = findDuplicates(streams.keySet(), this.streams.keySet());
+    checkArgument(duplicateStreams.isEmpty(),
+                  "Streams %s have been added already. Remove the duplicates.", duplicateStreams);
     this.streams.putAll(streams);
   }
 
   public void addDatasetSpecs(Map<String, DatasetCreationSpec> datasetSpecs) {
+    Set<String> duplicateDatasetSpecs = findDuplicates(datasetSpecs.keySet(), this.datasetSpecs.keySet());
+    checkArgument(duplicateDatasetSpecs.isEmpty(),
```

Comment:
This makes sense only if we are writing to same dataset from two different pipelines.